sql/core/src/test/scala/org/apache/spark/sql/sources/v2/TestInMemoryTableCatalog.scala

```diff
@@ -234,7 +234,8 @@ class InMemoryTable(
 
   private class Overwrite(filters: Array[Filter]) extends TestBatchWrite {
     override def commit(messages: Array[WriterCommitMessage]): Unit = dataMap.synchronized {
-      dataMap --= deletesKeys(filters)
+      val deleteKeys = InMemoryTable.filtersToKeys(dataMap.keys, partFieldNames, filters)
+      dataMap --= deleteKeys
```

Comment:
Is the refactoring in this file required? I like the refactor, but I don't think it is making a functional change so we should probably not include it in this commit.

Reply:
Nevermind, I see the uses in tests below.